app/src/features/surveys/view/survey-animals/device.ts

```diff
@@ -12,7 +12,11 @@ export type ITelemetryPointCollection = { points: FeatureCollection; tracks: Fea
 
 const req = 'Required.';
 const mustBeNum = 'Must be a number';
-const numSchema = yup.number().typeError(mustBeNum);
+const mustBePos = 'Must be positive';
+const mustBeInt = 'Must be an integer';
+const maxInt = 2147483647;
+const numSchema = yup.number().typeError(mustBeNum).min(0, mustBePos);
+export const intSchema = numSchema.max(maxInt, `Must be less than ${maxInt}`).integer(mustBeInt).required(req);
```

Comment:
Nit: if you are going to export this it should probably be made slightly more generic. But it doesn't look like it's used elsewhere so it probably doesn't need to be exported anyways.